pkg/controller/compliancescan/compliancescan_controller_test.go

```diff
@@ -51,15 +70,49 @@ var _ = Describe("Testing compliancescan controller phases", func() {
 			Expect(result).ToNot(BeNil())
 			Expect(err).To(BeNil())
 			Expect(compliancescaninstance.Status.Phase).To(Equal(complianceoperatorv1alpha1.PhaseRunning))
+
+			// We should have scheduled a pod per node
+			nodes, _ := getTargetNodes(&reconciler, compliancescaninstance)
+			var pods corev1.PodList
+			reconciler.client.List(context.TODO(), &pods)
+			Expect(len(pods.Items)).To(Equal(len(nodes.Items)))
 		})
 	})
 
 	Context("On the RUNNING phase", func() {
-		It("should update the compliancescan instance to phase DONE", func() {
-			result, err := reconciler.phaseRunningHandler(compliancescaninstance, logger)
-			Expect(result).ToNot(BeNil())
-			Expect(err).To(BeNil())
-			Expect(compliancescaninstance.Status.Phase).To(Equal(complianceoperatorv1alpha1.PhaseDone))
+		Context("With no pods in the cluster", func() {
+			It("should update the compliancescan instance to phase DONE", func() {
+				result, err := reconciler.phaseRunningHandler(compliancescaninstance, logger)
+				Expect(result).ToNot(BeNil())
+				Expect(err).To(BeNil())
+				Expect(compliancescaninstance.Status.Phase).To(Equal(complianceoperatorv1alpha1.PhaseDone))
+			})
+		})
+
+		Context("With two pods in the cluster", func() {
+			BeforeEach(func() {
+				// Create the pods for the test
+				reconciler.client.Create(context.TODO(), &corev1.Pod{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: fmt.Sprintf("%s-%s-pod", compliancescaninstance.Name, nodeinstance1.Name),
+					},
+				})
+				reconciler.client.Create(context.TODO(), &corev1.Pod{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: fmt.Sprintf("%s-%s-pod", compliancescaninstance.Name, nodeinstance2.Name),
+					},
+				})
+				// Set state to LAUNCHING
+				compliancescaninstance.Status.Phase = complianceoperatorv1alpha1.PhaseLaunching
+				reconciler.client.Status().Update(context.TODO(), compliancescaninstance)
+			})
+
+			It("should stay in LAUNCHING state", func() {
```

Comment:
I actually wonder if you found a bug here. Because in `isPodRunningInNode` we only check if the pod exists based on the name and if the pod exists, we treat it as running. But, shouldn't we also check the owner reference and if it doesn't point to a scan, just go to an error state?